dartagnan/src/main/java/com/dat3m/dartagnan/verification/model/event/CondJumpModel.java

```diff
@@ -0,0 +1,33 @@
+package com.dat3m.dartagnan.verification.model.event;
+
+import com.dat3m.dartagnan.program.Register.Read;
+import com.dat3m.dartagnan.program.event.Event;
+import com.dat3m.dartagnan.program.event.RegReader;
+import com.dat3m.dartagnan.program.event.core.CondJump;
+import com.dat3m.dartagnan.program.event.core.IfAsJump;
+import com.dat3m.dartagnan.verification.model.ThreadModel;
+
+import java.util.Set;
+import java.util.List;
+
+
+public class CondJumpModel extends DefaultEventModel implements RegReaderModel {
+    public CondJumpModel(CondJump event, ThreadModel thread, int id) {
+        super(event, thread, id);
+    }
+
+    public List<Event> getDependentEvents() {
+        if (((CondJump) event).isGoto() || ((CondJump) event).isDead()) {
+            return List.of();
+        } else if (event instanceof IfAsJump jump) {
+            return jump.getBranchesEvents();
+        } else {
+            return ((CondJump) event).getSuccessor().getSuccessors();
+        }
+    }
```

Comment:
This is used by the visitor of the control dependency which gets the event and them computed the EventModel. Why not to directly return the EventModels here?

Reply:
I fully agree. This method is not good. Not only does it return `Event`s but it also returns many that are not even part of the execution. This is problematic. Also, with the exception of `IfAsJump`, all events after the jump are control-flow dependent (i.e. all `EventModel`s that appear after the jump).